test/extended/olm/olmv1.go

```diff
@@ -99,6 +101,43 @@ var _ = g.Describe("[sig-olmv1][OCPFeatureGate:NewOLM][Skipped:Disconnected] OLM
 	})
 })
 
+var _ = g.Describe("[sig-olmv1][OCPFeatureGate:NewOLM][Skipped:Disconnected] OLMv1 Catalog's /v1/api/all endpoint", func() {
+	defer g.GinkgoRecover()
+	oc := exutil.NewCLIWithoutNamespace("default")
+
+	g.It("should serve FBC", func(ctx g.SpecContext) {
+		checkFeatureCapability(ctx, oc)
+
+		// Start port-forward in background
+		_, _, _, err := oc.AsAdmin().Run("port-forward").Args("-n", "openshift-catalogd", "svc/catalogd-service", "8080:443").Background()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		// Wait for port-forward to be established
+		time.Sleep(2 * time.Second)
+
+		g.By("Testing /api/v1/all endpoint for catalog openshift-community-operators")
+		verifyAPIEndpoint(ctx, "https://localhost:8080/catalogs/openshift-community-operators/api/v1/all")
+	})
+})
+
+var _ = g.Describe("[sig-olmv1][OCPFeatureGate:NewOLMCatalogdAPIV1Metas][Skipped:Disconnected] OLMv1 Catalog's /v1/api/metas endpoint", func() {
+	defer g.GinkgoRecover()
+	oc := exutil.NewCLIWithoutNamespace("default")
+	g.It(" should serve the /v1/api/metas API endpoint", func(ctx g.SpecContext) {
+		checkFeatureCapability(ctx, oc)
+
+		// Start port-forward in background
+		_, _, _, err := oc.AsAdmin().Run("port-forward").Args("-n", "openshift-catalogd", "svc/catalogd-service", "8443:443").Background()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		// Wait for port-forward to be established
+		time.Sleep(2 * time.Second)
+
+		g.By("Testing api/v1/metas endpoint for catalog openshift-community-operators")
+		verifyAPIEndpoint(ctx, "https://localhost:8443/catalogs/openshift-community-operators/api/v1/metas?schema=olm.package")
+	})
+})
+
```

Comment:
Is there a place to implement upgrade interruption tests? 

Another useful test to add (but not specific to the new metas feature gate):
1. Start a loop that on every iteration: gets a ClusterCatalog's status.URLs value and makes a query to a valid endpoint under that URL.
2. Do the upgrade

The expectation would be that the loop begins to get connection errors or 404 responses as the new pod starts up, but within a certain threshold (1m?) begins responding with 200 responses again.

Reply:
By upgrade do you mean `openshift` upgrade? 

Cursory look did not reveal anything of that nature in this repository, I'll have to research a little more. I know openshift does upgrade tests, I'm not sure if those tests are written in this repository.